cms/celery.py

```diff
@@ -25,29 +25,29 @@
 APP.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
 
 
-class Router(AlternateEnvironmentRouter):
-    """
-    An implementation of AlternateEnvironmentRouter, for routing tasks to non-cms queues.
+def route_task(name, args, kwargs, options, task=None, **kw):  # pylint: disable=unused-argument
     """
+    Celery-defined method allowing for custom routing logic.
 
-    @property
-    def alternate_env_tasks(self):
-        """
-        Defines alternate environment tasks, as a dict of form { task_name: alternate_queue }
-        """
-        # The tasks below will be routed to the default lms queue.
-        return {
-            'completion_aggregator.tasks.update_aggregators': 'lms',
-            'openedx.core.djangoapps.content.block_structure.tasks.update_course_in_cache': 'lms',
-            'openedx.core.djangoapps.content.block_structure.tasks.update_course_in_cache_v2': 'lms',
-        }
-
-    @property
-    def explicit_queues(self):
-        """
-        Defines specific queues for tasks to run in (typically outside of the cms environment),
-        as a dict of form { task_name: queue_name }.
-        """
-        return {
-            'lms.djangoapps.grades.tasks.compute_all_grades_for_course': settings.POLICY_CHANGE_GRADES_ROUTING_KEY,
-        }
+    If None is returned from this method, default routing logic is used.
+    """
+    # Defines alternate environment tasks, as a dict of form { task_name: alternate_queue }
+    alternate_env_tasks = {
+        'completion_aggregator.tasks.update_aggregators': 'lms',
+        'openedx.core.djangoapps.content.block_structure.tasks.update_course_in_cache': 'lms',
+        'openedx.core.djangoapps.content.block_structure.tasks.update_course_in_cache_v2': 'lms',
```

Comment:
Maybe not as part of this PR, but shouldn't this configuration be configuration, rather than code?

Reply:
True, having this and `explicit_queues` as Django settings would simplify the implementation here somewhat (and let more of the code for this routing function be shared between cms and lms.

Reply:
Okay, I will move it there and refactor this function.